### PR TITLE
VTX-3411: abort failed upload, update metrics

### PR DESCRIPTION
**Pull request by fsdvh**
We should always abort failed multipart uploads to avoid paying for the s3 storage, also we should count the number of written bytes only in case of success, as it shows a misleading pictures in a Grafana dashboard

### CODE REVIEW

File: ballista/executor/src/replicator/mod.rs [hunk 1]
```diff
@@ -148,37 +149,89 @@ fn serialize_batch(batch: RecordBatch) -> Result<bytes::Bytes, BallistaError> {
...
     mut reader: AsyncStreamReader<BufReader<Compat<File>>>,
-    object_store: Arc<dyn ObjectStore>,
 ) -> Result<(), BallistaError> {
-    let (_, mut upload) = object_store.put_multipart(destination).await.map_err(|e| {
-        REPLICATION_FAILURE
-            .with_label_values(&["create_writer"])
-            .inc();
-        BallistaError::General(format!("Failed to create object store writer: {:?}", e))
-    })?;
+    let written = std::sync::atomic::AtomicU64::new(0);
```
Comment (thinkharderdev) [inline ballista/executor/src/replicator/mod.rs:156]: This is just a local variable so you shouldn't need an atomic here. You can just use a `let mut written = 0;`

**Review by thinkharderdev — APPROVED** [review]
One small nit

File: ballista/executor/src/replicator/mod.rs (hunk 1 above)
Comment (fsdvh) [reply ballista/executor/src/replicator/mod.rs]: yes, you're right

Comment (fsdvh) [reply ballista/executor/src/replicator/mod.rs]: Removed

**Pull request merged**